client/components/data/query-site-comments/index.jsx

```diff
@@ -18,8 +18,8 @@ export class QuerySiteComments extends Component {
 		requestComments( this.props );
 	}
 
-	componentDidUpdate( { siteId } ) {
-		if ( siteId !== this.props.siteId ) {
+	componentDidUpdate( { siteId, status } ) {
+		if ( siteId !== this.props.siteId || status !== this.props.status ) {
 			requestComments( this.props );
```

Comment:
This is more stylistic, but I do prefer that we explicitly pass though only what we want. This probably won't be an issue here, but we've had cases in the pass where we pass through extra props to functions/other components, and it becomes unclear to the reader of what the signature actually looks like. Your call on this one though 😄